### ISSUE TITLE
Timeline: make sources more prominent

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#145232]
Capturing feedback from @Tyriar from today's UX call: the "..." can make it hard to discover multiple timeline sources. Idea would be to show a prominent filter icon with a dropdown like we do in problems or extensions view

![image](https://user-images.githubusercontent.com/900690/158629795-495a91d8-de39-4f98-8350-3a9c4329a7f1.png)

![image](https://user-images.githubusercontent.com/900690/158629835-26e44be5-3f1f-4a34-b32d-115540c21484.png)

![image](https://user-images.githubusercontent.com/900690/158629900-19e4dc57-35a5-4f0b-baab-8d1b04f7cbc5.png)

//cc @misolori @daviddossett

**Comment by daviddossett** [comment]
Here's an idea using the explicit filter action rather than the "..." menu. @misolori this may overlap with your freeform text filtering work. Not sure how the two might coexist yet. 

https://user-images.githubusercontent.com/25163139/158664230-39e56183-6f4d-4c3a-998e-dc2bc763f6f0.mp4

**Comment by bpasero** [comment]
Yes I like this. The goal imho should be to have a consistent UI language for when we filter a view. I am sure we have more locations throughout our UI that we should make sure are all aligned then 👍

**Comment by misolori** [comment]
For the icon, I think using the regular filter icon would work as we use that in other places. Though we do tend to use this as a filter option with an active state, not sure if that would apply here:

<img width="207" alt="CleanShot 2022-03-16 at 17 12 31@2x" src="https://user-images.githubusercontent.com/35271042/158711889-e9d3c23e-a595-4d1f-975e-aec9a8c3cc79.png">

<img width="341" alt="CleanShot 2022-03-16 at 17 12 51@2x" src="https://user-images.githubusercontent.com/35271042/158711925-5ce98e3f-648f-4d20-91df-35c7a946e5b5.png">

**Comment by bpasero** [comment]
This was easy enough and I think should be done before we ship local history so that folks can get rid of the local entries easily if annoying:
![Recording 2022-03-17 at 07 41 27](https://user-images.githubusercontent.com/900690/158751773-5e6ee801-2648-4d4e-a0c4-41713d076aa0.gif)

**Comment by bpasero** [comment]
**Verification:** you can change the visible timeline providers via the filter icon in the timeline view toolbar.

**Comment by daviddossett** [comment]
Looks great! @misolori and I were talking offline about how to indicate that the list is filtered since he's working on filtering across the product. One option could be having the filter icon button in a sort of "active" state if the list had 1+ items unchecked.

<img width="321" alt="CleanShot 2022-03-17 at 10 09 17@2x" src="https://user-images.githubusercontent.com/25163139/158855898-21483d23-a3d0-4a55-9149-40b07d5d7adf.png">


It could also be filled instead:

<img width="330" alt="CleanShot 2022-03-17 at 10 07 29@2x" src="https://user-images.githubusercontent.com/25163139/158855576-cfd5091f-9afb-4a69-b694-b64057d1bd5f.png">

This would attempt to be consistent with other filtered areas like the Problems view. It's slightly different given that it's an input, but leaving here for any further discussion:

<img width="469" alt="CleanShot 2022-03-17 at 10 06 15@2x" src="https://user-images.githubusercontent.com/25163139/158855262-e7abf8ea-4847-4f4c-bf14-b640842dd2ab.png">

**Comment by bpasero** [comment]
If we do this, I would want it in a way that it is the least "in my face" as possible because I expect a lot of users to disable the local history entries, at least as long as we didn't implement the grouping. Maybe we can have a minimal overlay over the icon to indicate activeness?

**Comment by daviddossett** [comment]
It might make more sense to have a two distinct icons for this like we do for pinned/unpinned. Forcing an icon button state to indicate a filter is applied feels fine in the context of the rich text inputs as pictured above. It doesn't feel quite right as I look at it in the context of a view actions toolbar.

Created new issue for this here: #145769

**Comment by bpasero** [comment]
Yeah another icon is fine for me, not sure I am a big fan of a permanent background though. For example the notification bell icon indicates that there are notifications in the center with a little overlay.
